x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/aggs/InferencePipelineAggregationBuilder.java

```diff
@@ -186,8 +197,9 @@ public InferencePipelineAggregationBuilder rewrite(QueryRewriteContext context)
         if (model != null) {
             return this;
         }
+
         SetOnce<LocalModel> loadedModel = new SetOnce<>();
-        context.registerAsyncAction((client, listener) -> {
+        BiConsumer<Client, ActionListener<?>> modelLoadAction = (client, listener) ->
```

Comment:
Ah, nice! That's what to use when in need to return more than a single thing. I'll use that onwards!